### PR TITLE
feat: Remove all aider remnants from codebase (ADV-0066)

### PR DESCRIPTION
**Pull request by movito**
## Summary

- **Delete 16 dead files**: 5 shell script templates, 3 runtime scripts, 8 investigation files
- **Update CLI** (`cli.py`): Remove aider checks from `init`/`check`/`evaluate`/`review`/`validate` commands, remove shell script rendering and creation
- **Update tests**: Remove `mock_aider_command` fixture, all `shutil.which("aider")` mocks, aider-specific test cases (9 tests removed, ~20 tests cleaned of aider mocks)
- **Update CI**: Remove `pip install aider-chat` step, `.aider.conf.yml` assertion, script checks
- **Update docs**: README, SETUP, QUICK_START, UPGRADE, TROUBLESHOOTING, AGENT_INTEGRATION, agent/workflow docs, Serena memories
- **Fix check() regression**: Added explicit `os.path.exists()` check for config file after removing script checks that previously masked the issue
- **Net change**: +505 / -6,626 lines

## Context

Follows ADV-0065 (PR #60) which replaced the Aider transport layer with LiteLLM. This PR removes all remaining traces of aider from the codebase so new users never encounter references to a dependency that no longer exists.

Historical references in ADRs, completed tasks, changelogs, and project history are intentionally preserved.

## Test plan

- [x] All 542 tests pass
- [x] `./scripts/core/ci-check.sh` passes (format, lint, pattern lint, tests)
- [x] Zero `shutil.which("aider")` in live code
- [x] Zero `pip install aider-chat` in CI
- [x] `grep -ri aider` on live code returns zero hits
- [ ] CI passes on GitHub
- [ ] Bot reviews clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes core CLI execution paths (`evaluate`, `review`, `validate`) and log selection/exit-code handling, which could affect workflow behavior across environments despite being largely cleanup.
> 
> **Overview**
> Removes the last Aider remnants by deleting legacy `.adversarial` scripts/templates and investigation artifacts, and cleaning up CI/docs references.
> 
> Rewires the CLI so `evaluate()`/`review()` delegate to `run_evaluator()` (built-in configs) and `validate()` runs the configured test command directly via `subprocess.run(shlex.split(...))`, with updated log-file selection based on evaluator `output_suffix`.
> 
> Updates tests to drop Aider-specific mocking/edge cases and instead mock `adversarial_workflow.evaluators.runner.run_evaluator`, adding coverage for missing built-in evaluators and runner failures; CI no longer installs `aider-chat` and the bash job now asserts the new init outputs rather than legacy scripts.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit caa21844dd6cb7aeb2c27faede93a6f87e4132ca. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: docs/guides/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -791,46 +695,25 @@ Acceptance: 6 xfailed tests in test_validation.py pass, exit code 0
...
-   sed -n '100,200p' tasks/TASK-001.md > task-excerpt.md
-   aider --read task-excerpt.md
+   # If > 1000 lines, split into smaller files
+   adversarial split tasks/TASK-001.md
    ```
 
+2. **Too many evaluations per task**
+   - Plan your evaluations before running them
+   - Use targeted evaluators for specific concerns
+
```
[the remark this answers is not in the document]
Comment (movito) [reply docs/guides/TROUBLESHOOTING.md:709]: Fixed in ac71a59: Added missing blank line before code block in TROUBLESHOOTING.md.

File: QUICK_START.md [hunk 2]
```diff
@@ -468,13 +446,11 @@ EOF
...
-# 2. Implement with aider
-aider --architect-mode src/api.py
-
-# ... use aider to implement ...
+# 2. Implement with your preferred tool
+# ... use Claude Code, Cursor, or any editor ...
 
 # 3. Independent review (different AI)
-adversarial review  # Catches what aider might miss
+adversarial review  # Catches what you might miss
```
[the remark this answers is not in the document]
Comment (movito) [reply QUICK_START.md:454]: Fixed in ac71a59: Renamed Aider heading to Full Adversarial Review and updated example command with task_file argument.

File: SETUP.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ This guide walks you through setting up adversarial-workflow for development and
 
 Before you begin, ensure you have:
 
-1. **Python 3.11+** installed (required for aider compatibility)
+1. **Python 3.10+** installed
    ```bash
    python3 --version  # Should be 3.11 or higher
    ```
```
[the remark this answers is not in the document]
Comment (movito) [reply SETUP.md:14]: Fixed in ac71a59: Updated version hint from 3.11 to 3.10 to match documented minimum.

File: .github/workflows/test-package.yml [hunk 4]
```diff
@@ -194,12 +170,14 @@ jobs:
...
-            bash -n "$script" || exit 1
-          done
-          echo "✓ All scripts compatible with Linux bash"
+          # Verify initialization created expected files
+          test -d .adversarial || exit 1
+          test -f .adversarial/config.yml || exit 1
+          test -f .env.example || exit 1
+
+          # Verify no legacy script directory is created
+          test ! -d .adversarial/scripts || echo "Note: scripts directory exists but is no longer required"
```
[the remark this answers is not in the document]
Comment (movito) [reply .github/workflows/test-package.yml:179]: Fixed in ac71a59: Great catch. Replaced shell script execution in evaluate, review, and validate with LiteLLM-based run_evaluator calls. Also removed dead script-check code from health. Tests updated.

File: adversarial_workflow/cli.py [hunk 5]
```diff
@@ -1955,113 +1745,32 @@ def evaluate(task_file: str) -> int:
...
-            print("   This will reinstall all workflow scripts")
+    builtin_config = BUILTIN_EVALUATORS.get("evaluate")
+    if builtin_config is None:
+        print(f"{RED}❌ ERROR: Built-in 'evaluate' evaluator not found{RESET}")
         return 1
 
-    # Error 5: Evaluation rejected
-    if result.returncode != 0:
+    eval_result = run_evaluator(builtin_config, task_file)
+    if eval_result != 0:
```
[the remark this answers is not in the document]
Comment (movito) [reply adversarial_workflow/cli.py:1728]: Fixed in 0149317: Removed duplicate large-file gate from evaluate() — run_evaluator() already handles file size checks.

File: adversarial_workflow/cli.py [hunk 6]
```diff
@@ -1955,113 +1745,32 @@ def evaluate(task_file: str) -> int:
...
+    log_dir = config.get("log_directory", ".adversarial/logs/")
+    import glob
 
-    log_file = os.path.join(config["log_directory"], f"{task_num}-PLAN-EVALUATION.md")
+    log_files = sorted(glob.glob(os.path.join(log_dir, "*.md")), key=os.path.getmtime, reverse=True)
+    if not log_files:
+        print(f"{YELLOW}⚠️  No evaluation log found in {log_dir}{RESET}")
+        return 0
+
+    log_file = log_files[0]
```
[the remark this answers is not in the document]
Comment (movito) [reply adversarial_workflow/cli.py]: Fixed in 0149317: Now filters log files by evaluator output_suffix instead of newest *.md glob for reliable log selection.

File: adversarial_workflow/cli.py [hunk 7]
```diff
@@ -2210,24 +1912,19 @@ def validate(test_command: str | None = None) -> int:
...
-        print(f"{RED}❌ ERROR: Script not found: {script}{RESET}")
-        print("   Fix: Run 'adversarial init'")
-        return 1
-
+    # Run test command directly (no shell script needed)
     try:
-        result = subprocess.run([script, test_command], timeout=600)  # 10 minutes for tests
+        result = subprocess.run(
+            test_command.split(),
+            timeout=600,  # 10 minutes for tests
```
[the remark this answers is not in the document]
Comment (movito) [reply adversarial_workflow/cli.py:1898]: Fixed in 0149317: Changed to shlex.split(test_command) for proper tokenization of quoted arguments.

File: docs/guides/TROUBLESHOOTING.md [hunk 8]
```diff
@@ -508,62 +486,7 @@ Error: Request too large for model
...
-   ls -la tasks/TASK-001.md
-   ```
-
-3. **Check task_directory in config**
-   ```bash
-   # Config should match your project structure
-   cat .adversarial/config.yml
-   # task_directory: tasks/  # Must match actual location
-   ```
+## Evaluation Issues
```
[the remark this answers is not in the document]
Comment (movito) [reply docs/guides/TROUBLESHOOTING.md:489]: Fixed in 0149317: Updated TOC anchor from #script-execution-issues to #evaluation-issues.

File: docs/guides/TROUBLESHOOTING.md [hunk 9]
```diff
@@ -593,49 +516,30 @@ $ adversarial review
    # Review the last commit instead
    git diff HEAD~1 > .adversarial/artifacts/last-commit.diff
 
-   # Manually review using aider
-   aider --read .adversarial/artifacts/last-commit.diff \
-         --message "Review this implementation"
+   # Review using the CLI
+   adversarial review
```
[the remark this answers is not in the document]
Comment (movito) [reply docs/guides/TROUBLESHOOTING.md]: Fixed in 0149317: Updated review workaround to show correct `adversarial review task.md` usage with branch-aware diff explanation.

File: SETUP.md [hunk 10]
```diff
@@ -8,9 +8,9 @@ This guide walks you through setting up adversarial-workflow for development and
 
 Before you begin, ensure you have:
 
-1. **Python 3.11+** installed (required for aider compatibility)
+1. **Python 3.10+** installed
    ```bash
-   python3 --version  # Should be 3.11 or higher
+   python3 --version  # Should be 3.10 or higher
```
[the remark this answers is not in the document]
Comment (movito) [reply SETUP.md:13]: Fixed in 0149317: Changed `python3.11 -m venv .venv` to `python3 -m venv .venv` for consistency with 3.10+ requirement.

File: adversarial_workflow/cli.py [hunk 11]
```diff
@@ -2161,29 +1847,22 @@ def review(task_file: str) -> int:
...
-    try:
-        result = subprocess.run([script, task_file], timeout=180)
-    except subprocess.TimeoutExpired:
-        print(f"{RED}❌ ERROR: Review timed out (>3 minutes){RESET}")
+    builtin_config = BUILTIN_EVALUATORS.get("review")
+    if builtin_config is None:
+        print(f"{RED}❌ ERROR: Built-in 'review' evaluator not found{RESET}")
+        print("   Install a review evaluator: adversarial library install <name>")
+        print("   Or use: adversarial <evaluator-name> <task-file>")
         return 1
```
[the remark this answers is not in the document]
Comment (movito) [reply adversarial_workflow/cli.py:1859]: Won't fix — false positive. BUILTIN_EVALUATORS registers the key as 'review' (builtins.py line 226), not 'code-review'. The lookup is correct.

File: tests/test_evaluate.py [hunk 12]
```diff
@@ -42,265 +41,119 @@ def test_evaluate_config_load_error(self, mock_load_config, tmp_path, capsys):
...
+        log_file.write_text("# Evaluation\nVerdict: NEEDS_REVISION")
 
-        # Mock needs revision validation
-        mock_validate.return_value = (True, "NEEDS_REVISION", "Plan needs work")
+        mock_load_config.return_value = {"log_directory": str(log_dir) + "/"}
 
-        result = evaluate(str(task_file))
+        with patch("adversarial_workflow.evaluators.runner.run_evaluator", return_value=0):
+            mock_validate.return_value = (True, "NEEDS_REVISION", "Plan needs work")
+            result = evaluate(str(task_file))
```
[the remark this answers is not in the document]
Comment (movito) [reply tests/test_evaluate.py]: Fixed in 782acb5: Updated test to mock run_evaluator returning 1 (the actual exit code for NEEDS_REVISION) and verify evaluate() returns early with non-zero exit.

File: tests/test_evaluate.py [hunk 13]
```diff
@@ -42,265 +41,111 @@ def test_evaluate_config_load_error(self, mock_load_config, tmp_path, capsys):
...
 
-        # Mock successful validation
-        mock_validate.return_value = (True, "APPROVED", "Plan approved")
+        mock_load_config.return_value = {"log_directory": str(log_dir) + "/"}
 
-        result = evaluate(str(task_file))
+        # Mock run_evaluator to succeed
+        with patch("adversarial_workflow.evaluators.runner.run_evaluator", return_value=0):
+            mock_validate.return_value = (True, "APPROVED", "Plan approved")
+            result = evaluate(str(task_file))
```
[the remark this answers is not in the document]
Comment (movito) [reply tests/test_evaluate.py:84]: Won't fix — false positive. The import is a local import inside the function body (lazy import), so patching at the source module works correctly. All 14 tests pass.

File: tests/test_evaluate.py [hunk 14]
```diff
@@ -42,265 +41,111 @@ def test_evaluate_config_load_error(self, mock_load_config, tmp_path, capsys):
...
-            patch(
-                "adversarial_workflow.cli.load_config",
-                return_value={"log_directory": ".adversarial/logs/"},
-            ),
-            patch("os.path.exists", return_value=True),
-            patch("builtins.input", return_value="n"),
-        ):  # User says no
-            result = evaluate(str(task_file))
-            assert result == 0  # Cancelled, not error
+        assert result == 0
```
[the remark this answers is not in the document]
Comment (movito) [reply tests/test_evaluate.py:152]: Fixed in caa2184: Now provides a real log file instead of mocking glob.glob to return [], so the validation path is actually exercised.

File: tests/test_evaluate.py [hunk 15]
```diff
@@ -444,20 +289,24 @@ def test_verify_token_count_low_warning(self, mock_extract, mock_estimate, tmp_p
...
+                return_value=0,
+            ),
             patch(
                 "adversarial_workflow.cli.validate_evaluation_output",
                 return_value=(True, "APPROVED", "OK"),
             ),
             patch("adversarial_workflow.cli.verify_token_count"),
+            patch("glob.glob", return_value=[]),
         ):
             result = evaluate(str(sample_task_file))
```
[the remark this answers is not in the document]
Comment (movito) [reply tests/test_evaluate.py]: Fixed in caa2184: Same glob mock bypass fix applied to the integration test.
